### PR TITLE
add is:extraperk filter

### PR DESCRIPTION
**Pull request by nev-r**
highlights randomized weapons with extra perks

the main focus here is their rarity/prestige, and the limited time you're able to acquire them.
limiting to legendaries w/ randomized sockets causes the filter to exclude
- pinnacle/pursuit weapons, which always come with fixed selectable perks
- exotics like suros (are there other exotics like suros?)
- y1 ikelos weapons

and include:
- adept/timelost weapons, whose column(s) have a random perk, plus one that's always the same
- "additional perk" umbral focusings, from arrivals and later

the wording for this feature in-game seems to usually be "additional perk" but frankly that's a long word i don't like it

### CODE REVIEW

**Review by delphiactual — CHANGES_REQUESTED** [review]
update tests

**Review by delphiactual — APPROVED** [review]

**Comment by nev-r** [comment]
yep :grinning:

File: src/app/search/search-filters/sockets.tsx [hunk 1]
```diff
@@ -87,6 +88,24 @@ const socketFilters: FilterDefinition[] = [
...
+    destinyVersion: 2,
+    filter: () => (item: DimItem) => {
+      if (!(item.bucket?.sort === 'Weapons' && item.tier.toLowerCase() === 'legendary')) {
+        return false;
+      }
+
+      return getSocketsByCategoryHash(item.sockets, SocketCategoryHashes.WeaponPerks_Reusable)
+        .filter(
+          (socket) =>
+            socket.plugged?.plugDef.plug.plugCategoryIdentifier === 'frames' &&
```
Comment (robojumper) [inline src/app/search/search-filters/sockets.tsx:103]: `plugCategoryHash === PlugCategoryHashes.Frames`?

**Review by robojumper — APPROVED** [review]

File: src/app/search/search-filters/sockets.tsx (hunk 1 above)
Comment (nev-r) [reply src/app/search/search-filters/sockets.tsx:103]: indeed

**Pull request merged**